hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DistributedFileSystem.java

```diff
@@ -3862,6 +3869,11 @@ protected EnumSet<CreateFlag> getFlags() {
      */
     @Override
     public FSDataOutputStream build() throws IOException {
+      if (getOptionalKeys().contains(
+          Options.OpenFileOptions.FS_OPTION_OPENFILE_EC_POLICY)) {
+        ecPolicyName(getOptions().get(
```

Comment:
Just do 
```
policy = getOptions().get(FS_OPTION_OPENFILE_EC_POLICY, "")
if (!policy.isEmpty()) {
 ecPolicyName(policy)
}
```
no need to look for the keys, just look up the value and act only on whether or not it was present

Reply:
Done.